Add wrapper submodule

ValRat opened the pull request:
Rosinstall used as a source file grabber, fixes #304 
added realsense camera module, fixes #302

garethellis0 commented on .rosinstall:
```diff
@@ -3,10 +3,10 @@
-- git: {local-name: external_pkgs/zed-ros-wrapper, uri: 'https://github.com/UBC-Snowbots/zed-ros-wrapper.git'}
-- git: {local-name: external_pkgs/ros_arduino_bridge, uri: 'https://github.com/UBC-Snowbots/ros_arduino_bridge.git'}
-- git: {local-name: external_pkgs/nmea_navsat_driver, uri: 'https://github.com/UBC-Snowbots/nmea_navsat_driver.git'}
+- setup-file: {local-name: /opt/ros/kinetic/setup.sh}
```
What does changing this line do? @Dulluhan?

Dulluhan replied:
In theory, ROS will not know how to link the setup.sh under `external_pkgs` to the ros stack. But with recent findings, I am not sure anymore. Will need to check

ValRat replied:
Digging into it, it seems that it doesn't matter too much cause [wstool](http://wiki.ros.org/wstool) ignores this anyways per [REP-126](http://www.ros.org/reps/rep-0126.html).
Also, the hottest thing in town is `wstool` since it is the successor to `rosws` which succeeds `rosinstall`.

ValRat replied:
The devs also do admit `rosinstall` was incredibly powerful but had confusing commands and external effects. Its iterations are made to be simpler with commands having clear single responsibilities.

ValRat commented:
Good for re-review, @Dulluhan @garethellis0.

Added separate setup script for realsense2 related packages and fixed small leftover bugs from migrating the setup scripts into their own folders.

ValRat commented on setup_scripts/setup_realsense.sh:
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
This file is called within `install_dependencies.sh`.

ValRat commented on setup_scripts/install_dependencies.sh:
```diff
@@ -45,20 +50,17 @@ echo "================================================================"
 
 # Setup udev rules
-sudo cp extended_pkg/phidgets_api/share/udev/99-phidgets.rules /etc/udev/rules.d
+sudo cp $CURR_DIR/../src/extended_pkgs/phidgets_api/share/udev/99-phidgets.rules /etc/udev/rules.d
```
It's a little bit awkward cause this has a hard dependency on `rosinstall` to pull in the `.rules` file for phidgets. I'm OK with this being here even though putting it with the arduino udev rules would make it cleaner.

ValRat commented:
Made another change by migrating to `wstool`. Last change, I promise @Dulluhan @garethellis0.

garethellis0 commented:
No worries! I'll never complain about getting a second review request because we've found a better way to do something! :smile:

garethellis0 commented on setup_scripts/setup_realsense.sh:
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
It would be helpful to have a quick comment here explaining what this file is for (it's association with the realsense package, why we need it, etc.)

ValRat replied:
Added

garethellis0 commented on setup_scripts/install_tools.sh:
```diff
@@ -26,13 +26,11 @@ SHELL_CONFIG_FILES=(
-    "export GAZEBO_RESOURCE_PATH=$DIR/src/sb_gazebo/models:${GAZEBO_RESOURCE_PATH}"\
+    "export GAZEBO_MODEL_PATH=$DIR/../src/sb_gazebo/models:${GAZEBO_MODEL_PATH}"\
+    "export GAZEBO_PLUGIN_PATH=$DIR/../src/sb_gazebo/lib:${GAZEBO_PLUGIN_PATH}"\
+    "export GAZEBO_RESOURCE_PATH=$DIR/../src/sb_gazebo/models:${GAZEBO_RESOURCE_PATH}"\
```
Whoops, good catch this, my bad for missing that when I changed setup scripts directory.

garethellis0 commented:
Looks good! :+1: :tada:
